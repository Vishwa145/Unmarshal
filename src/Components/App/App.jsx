import React from "react";
import Header from "../Header/Header";
import Body from "../Body/Body";

class App extends React.Component {
  constructor(props) {
    super(props);
    //Maintian the error status.
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {

    return (
      <div className="App">
        {this.state.hasError === false ? <React.Fragment><Header />
            <Body />
            </React.Fragment>
            : <h1>Something broke, please close and try again.</h1>}
      </div>
    );
  }
}

export default App;
