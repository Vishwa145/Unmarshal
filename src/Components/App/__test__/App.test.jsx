import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import renderer from "react-test-renderer"

it("render successfully", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Matches snapshot", ()=>{
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
