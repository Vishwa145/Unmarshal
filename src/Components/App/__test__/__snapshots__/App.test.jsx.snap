// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="Nav"
  >
    <div
      className="Logo"
    >
      <img
        alt="Logo"
        src="logo.png"
      />
      <h2>
        unmarshal
      </h2>
    </div>
    <div
      className="SearchBar"
    >
      <input
        name="email"
        placeholder="Search by Address / Txn Hash"
        type="text"
      />
      <button>
        <img
          alt=""
          height="100%"
          src="SearchIcon.svg"
        />
      </button>
    </div>
  </div>
  <div
    className="Body"
  >
    <div
      className="Summary"
    >
      <h3>
        <strong>
          Transaction Details
        </strong>
      </h3>
      <p>
        Unmarshal - sponsered slots available 
        <a
          href="/"
        >
          Book your slot here!
        </a>
      </p>
    </div>
    <div
      className="Container"
    >
      <div
        className="SwitchTab"
      >
        <button>
          <h3>
            Ethereum
          </h3>
        </button>
        <button>
          <h3>
            BSC
          </h3>
        </button>
        <button>
          <h3>
            Ethereum
          </h3>
        </button>
      </div>
      <h1>
        Loading
        ...
      </h1>
    </div>
  </div>
</div>
`;
